cardano-node/src/Cardano/Node/Handlers/TopLevel.hs

```diff
@@ -91,6 +93,17 @@ toplevelExceptionHandler prog = do
     rethrowExitCode :: ExitCode -> IO a
     rethrowExitCode = throwIO
 
+    rethrowDiffusionErrorExceptionInLinkedThreadExitSuccess :: Network.Failure -> IO a
+    rethrowDiffusionErrorExceptionInLinkedThreadExitSuccess full =
+      case full of
+        Network.DiffusionError e ->
+          case fromException (toException e) of
+            Just (ExceptionInLinkedThread _ eInner)
+              | Just ExitSuccess <- fromException eInner
+              -> throwIO ExitSuccess
```

Comment:
```suggestion
              | Just err <- fromException eInner :: Maybe ExitCode
              -> throwIO err
```
I think it's better to treat any constructor of `ExitCode` in the same way.

Reply:
https://github.com/input-output-hk/cardano-node/pull/5243/files#r1211703792